pfsimulator/parflow_lib/problem_wc_y.c

```diff
@@ -0,0 +1,471 @@
+/*BHEADER**********************************************************************
+*
+*  Copyright (c) 1995-2024, Lawrence Livermore National Security,
+*  LLC. Produced at the Lawrence Livermore National Laboratory. Written
+*  by the Parflow Team (see the CONTRIBUTORS file)
+*  <parflow@lists.llnl.gov> CODE-OCEC-08-103. All rights reserved.
+*
+*  This file is part of Parflow. For details, see
+*  http://www.llnl.gov/casc/parflow
+*
+*  Please read the COPYRIGHT file or Our Notice and the LICENSE file
+*  for the GNU Lesser General Public License.
+*
+*  This program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License (as published
+*  by the Free Software Foundation) version 2.1 dated February 1999.
+*
+*  This program is distributed in the hope that it will be useful, but
+*  WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms
+*  and conditions of the GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU Lesser General Public
+*  License along with this program; if not, write to the Free Software
+*  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+*  USA
+**********************************************************************EHEADER*/
+
+#include "parflow.h"
+#include "parflow_netcdf.h"
+
+
+/*--------------------------------------------------------------------------
+ * Structures
+ *--------------------------------------------------------------------------*/
+
+typedef struct {
+  int type;
+  void  *data;
+  int wcy_exists;              /* check to see whether or not channel width Y is used */
+} PublicXtra;
+
+typedef struct {
+  Grid   *grid3d;
+  Grid   *grid2d;
+} InstanceXtra;
+
+typedef struct {
+  NameArray regions;
+  int num_regions;
+  int     *region_indices;
+  double  *values;
+} Type0;                       /* constant regions */
+
+typedef struct {
+  char  *filename;
+  Vector *wcy_values; 
+} Type1;                       /* .pfb file */
+
+typedef struct {
+  char  *filename;
+  Vector *wcy_values;
+} Type2;                       /* .nc file */
+
+/*
+* YChannelWidth
```

Comment:
Methods should be documented using Doxygen format. See

https://github.com/parflow/parflow/blob/master/CONTRIBUTING.md#source-code-documentation

grgeom_octree.h has Doxygen comments as examples.